src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -167,4 +177,25 @@ def post(self, request, *args, **kwargs):
             return HttpResponseRedirect(reverse("admin:catalogi_catalogus_changelist"))
         except (CommandError, IntegrityError) as exc:
             messages.add_message(request, messages.ERROR, exc)
+
+        catalogus_pk = kwargs.get("catalogus_pk")
+        catalogus = Catalogus.objects.get(pk=catalogus_pk)
+
+        if iotype_forms:
+            for i, form in zip(request.session["iotypen"], iotype_forms.forms):
```

Comment:
This feels kinda hacky but otherwise it I would need to iterative over the labels above. Not sure if this actually improves performance.